db/appsock_handler.c

```diff
@@ -265,6 +268,33 @@ static void appsock_work(struct thdpool *pool, void *work, void *thddata)
     struct appsock_thd_state *state = thddata;
     appsock_work_args_t *w = (appsock_work_args_t *)work;
     int keepsocket = 0;
+    int curr_appsock_conns = 0;
+    w->rejected = 0;
+    int num_retry = 0;
+
+retry:
+    Pthread_mutex_lock(&appsock_conn_lk);
+    if (active_appsock_conns >= bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT)) {
+        if (close_lru_client(active_appsock_conns, NULL, bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT)) != 0) {
```

Comment:
close_lru_client doesn't actually close - it signals the appsock thread to close.  So even if we found a suitable connection, we still end up allowing more than the maximum number during a spike.

Reply:
Yes, the thread doesn't close immediately, however number of actual open connections decreases